engine/src/main/resources/application.conf

```diff
@@ -36,6 +36,9 @@ system {
 
   // If 'true', a SIGINT will trigger Cromwell to attempt to abort all currently running jobs before exiting
   abort-jobs-on-terminate = false
+
+  // Max number of retries per job that the engine will attempt in case of a retryable failure received from the backend
+  maxRetries = 3
```

Comment:
My opinion here was to have a decent enough default value in the engine side to accommodate our current use cases (current value of `3` is just a start.. we can and should probably increase it). But relying solely on the backend doesn't sound like a very good idea in the world of 3rd party developers. This can later (quite later) on also have the added benefit that after certain number of retries, if the job still fails, we can schedule it on a different backend maybe.